app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModel.kt

```diff
@@ -276,6 +282,32 @@ class ConversationMessagesViewModel @Inject constructor(
         }
     }
 
+    fun updateImageOnFullscreenMode(message: UIMessage?) {
+        imageMessageOnFullscreen = message
+    }
+
+    fun checkPendingActions(onMessageReply: (UIMessage) -> Unit) = viewModelScope.launch {
```

Comment:
for me personally this function is hard to follow what is exactly happening, perhaps it is because I did not work with savedStateHandle but just from reading this I have no idea what is going on